frontend/src/utils/GeneralUtils.ts

```diff
@@ -94,3 +99,16 @@ export const getLetterFromNumber = (number: number): string => {
 
   return firstLetter + secondLetter;
 };
+
+export const preventNonNumericKeys = (
+  e: React.KeyboardEvent<HTMLInputElement>,
+  value: string,
+  integerOnly = false,
+) => {
+  if (
+    (integerOnly && e.key === ".") ||
+    e.key === "e" ||
+    (e.key === "-" && value.length !== 0)
```

Comment:
this doesn't handle the case of a user first inputting "123", then adding the sign after "-123". we can further look into it in this ticket: https://www.notion.so/uwblueprintexecs/Allow-hyphen-in-Short-Answer-and-Fraction-at-start-of-input-4334fe3c414241c983886bd8ccf186eb?pvs=4

Reply:
sounds good!